proposals/2023-08-21-utf8.md

```diff
@@ -0,0 +1,258 @@
+
+## How
+
+### But First: Differences between PromQL and Openmetrics Exposition
+
+Changes will need to be made to PromQL and the exposition format. It is important to note that while these two languages have surface similarities, there are important differences between them that need to be accounted for:
+
+* Quotes:
+  * **PromQL**: Strings may be quoted with single or double quotes. Strings may also be quoted with backticks to avoid escape sequences (raw strings).
+  * **Exposition format**: Supports only double quotes.
+* Equal Sign:
+  * **PromQL**: The equal sign (`=`) is a label matching operator, and there are other label matching operators for other operations.
+  * **Exposition Format**: The equal sign is an assignment operator that sets a label to a specific value.
+* String Escaping:
+  * **PromQL**: Strings are unquoted using the Go strconv.Unquote function.
+  * **Exposition Format**: Characters are escaped with a custom backslash format.
+
+### What will the new syntax look like?
+
+**tl;dr:**
+
+`{"my.noncompliant.metric", "label"="value"}`
+
+#### Oh no, I hate it
+
+A full discussion of rejected alternatives can be found below, but a common first reaction is, why move the metric name inside the braces? Why not leave it outside, like: `"my.noncompliant.metric"{"label"="value"}`.
+
+There are some specific risks with this format, but we also wanted to weigh the value of a syntax that is **more familiar to current experts** vs **more approachable for new users**. We take the optimistic position that there are more potential new users of Prometheus than there are existing users, and therefore decided to give more weight to that framing.
+
+The previous syntax makes sense because it resembles a function call, the way that `foo{bar}` is like `foo(bar)`. The metric name is the primary selector, and the braces enclose parameters on that selection. If we were to add the additional quote characters, this frame of reference is lost. For new users, visually parsing the query, with brace in the middle of a noisy series of quoted values, is likely to be confusing.
+
+Instead, we believe that it is more intuitive and easier to teach new users to think of the brace as **the characters that surround the parameters for selecting data**, where **the "metric name" is a special case** of that kind of selector. This is similar to the approach chosen for [LogQL](https://grafana.com/docs/loki/latest/query/) and [TraceQL](https://grafana.com/docs/tempo/latest/traceql/), which do not have a special case "metric name" equivalent and thus have no items outside the curly braces.
+
+Furthermore, PromQL users will already know that it's possible to put the metric name inside the braces, for the purpose of using an operator to enable richer metric selection, like: `{__name__=~"my\.metric.*"}`. The chosen syntax serves as a shortening of this operation, and is forward-compatible with future metric matching prefix operators if we decide to implement them.
```

Comment:
`{__name__=~"my\.metric.*"}` results in `parse error: unknown escape sequence U+002E`. Since PromQL query are first run through Go escaping (as explained in this doc ;), this has to be `{__name__=~"my\\.metric.*"}` or alternatively ``{__name__=~`my\.metric.*`}``.